Src/Plugins/Security/SEC101_001.HttpAuthorizationRequestHeaderValidator.cs

```diff
@@ -60,9 +61,10 @@ public static ValidationState IsValidDynamic(ref Fingerprint fingerprint, ref st
             fingerprint = new Fingerprint
             {
                 Host = host,
+                Scheme = scheme,
                 Secret = secret,
-                Resource = (groups.ContainsKey("resource") && groups["resource"] != "/")
-                    ? groups["resource"]
+                Path = (groups.ContainsKey("path") && groups["path"] != "/")
```

Comment:
>= [](start = 21, length = 1)

I think our pattern for this would be to have an unchecked 

groups.TryGetNonEmptyValue("path", out string path);

then just assign path to this property in the fingerprint. Cleaner.

Reply:
groups.TryGetNonEmptyValue("path", out string path);

// Do more processing of path argument here to put it into a null vs. non-null condition.

fingerprint = new Fingerprint
            {
                Host = host,
                Scheme = scheme,
                Secret = secret,
                Path = path;
            };




---
In reply to: [612765053](https://github.com/microsoft/sarif-pattern-matcher/pull/380#discussion_r612765053) [](ancestors = 612765053)